osu.Game/Beatmaps/IO/LegacyFilesystemReader.cs

```diff
@@ -26,7 +26,7 @@ public override void Dispose()
             // no-op
         }
 
-        public override IEnumerable<string> Filenames => Directory.GetFiles(path).Select(Path.GetFileName).ToArray();
+        public override IEnumerable<string> Filenames => Directory.GetDirectories(path).Select(Path.GetFileName).ToArray();

```

Comment:
Pretty sure this is not what you want as this will only give you directory paths. You want to instead be doing something like 
```
Directory.GetFiles(path)
    .Concat(Directory.GetDirectories(path).SelectMany(d => Directory.GetFiles(d)))
    .Select(Path.GetFileName).ToArray();
```

Reply:
don't we need to recursively get all files from subdirectories on top of this?

Reply:
Good catch, in that case it'd be even just `Directory.GetFiles(path, "*.*", SearchOption.AllDirectories)`.